drivers/hmis/app/graphql/mutations/create_inventory.rb

```diff
@@ -6,12 +6,91 @@ class CreateInventory < BaseMutation
     field :errors, [Types::HmisSchema::ValidationError], null: false
 
     def resolve(input:)
-      default_create_record(
+      response = default_create_record(
         Hmis::Hud::Inventory,
         field_name: :inventory,
         id_field_name: :inventory_id,
         input: input,
       )
+
+      inventory = response[:inventory]
+      return response unless inventory.present?
+
+      create_beds_and_units(inventory, input.beds_per_unit)
+
+      response
+    end
+
+    def create_beds_and_units(inventory, beds_per_unit)
+      return unless inventory.unit_inventory.positive?
+
+      common = { user_id: hmis_user.user_id, created_at: Time.now, updated_at: Time.now }
+
+      # Create Units
+
+      unit_args = (1..inventory.unit_inventory).map do |i|
+        {
+          inventory_id: inventory.id,
+          name: "Unit #{i}",
+          **common,
+        }
+      end
+      units = Hmis::Unit.insert_all(unit_args)
+
+      # Create Beds
+
+      bed_args = []
+      unit_idx = 0
+      beds_filled = 0
+      Hmis::Bed.bed_types.each do |bed_type|
+        num_beds = inventory.send(bed_type)
+        next unless num_beds&.positive?
+
+        (1..num_beds).each do |i|
+          # Try to fill beds according to beds_per_unit.
+          # If we have too many beds, puts all the remaining ones in the last unit.
```

Comment:
I wonder if we shouldn't do this since beds need types, and I don't think there's any reason they would be evenly distributed across units or necessarily grouped together.